Vic2ToHoI4/Source/V2World/World/WorldFactory.cpp

```diff
@@ -353,4 +354,77 @@ void Vic2::World::Factory::consolidateConquerStrategies()
 	{
 		country->consolidateConquerStrategies(world->provinces);
 	}
+}
+
+
+void Vic2::World::Factory::resolveBattles()
+{
+	const auto armyLocations = determineArmyLocations();
+
+	for (auto& [location, armies]: armyLocations)
+	{
+		if (armies.size() == 1)
+		{
+			continue;
+		}
+		if (!world->provinces.contains(location)) // true of sea provinces
+		{
+			continue;
+		}
+		if (!armiesHaveDifferentOwners(armies))
+		{
+			continue;
+		}
+
+		const auto& provinceOwner = world->provinces.at(location)->getOwner();
+		for (const auto& army: armies)
+		{
+			if (army->getOwner() != provinceOwner)
+			{
+				army->setLocation(std::nullopt);
+			}
+		}
+	}
+}
+
+
+std::map<int, std::vector<Vic2::Army*>> Vic2::World::Factory::determineArmyLocations() const
```

Comment:
Vector of raw pointers? I assume a necessity for some reason.

Reply:
Yeah, best of bad choices. I didn't want to copy the armies, since I need to change the originals. Can't put a reference in a container. And using shared_ptr would imply the ownership is shared when it's not really. Thus, raw pointers.